askama_parser/src/lib.rs

```diff
@@ -599,4 +686,37 @@ mod test {
         assert_eq!(num_lit("1_.").unwrap(), (".", "1_"));
         assert_eq!(num_lit("1_2.").unwrap(), (".", "1_2"));
     }
+
+    #[test]
+    fn test_char_lit() {
+        assert_eq!(char_lit("'a'").unwrap(), ("", "a"));
+
+        // Escaped single characters.
+        assert_eq!(char_lit("'\\\"'").unwrap(), ("", "\\\""));
+        assert_eq!(char_lit("'\\''").unwrap(), ("", "\\'"));
+        assert_eq!(char_lit("'\\t'").unwrap(), ("", "\\t"));
+        assert_eq!(char_lit("'\\n'").unwrap(), ("", "\\n"));
+        assert_eq!(char_lit("'\\r'").unwrap(), ("", "\\r"));
+        assert_eq!(char_lit("'\\0'").unwrap(), ("", "\\0"));
+        // Escaped ascii characters (up to `0x7F`).
+        assert_eq!(char_lit("'\\x12'").unwrap(), ("", "\\x12"));
+        assert_eq!(char_lit("'\\x02'").unwrap(), ("", "\\x02"));
+        assert_eq!(char_lit("'\\x6a'").unwrap(), ("", "\\x6a"));
+        assert_eq!(char_lit("'\\x7F'").unwrap(), ("", "\\x7F"));
+        // Escaped unicode characters (up to `0x10FFFF`).
+        assert_eq!(char_lit("'\\u{A}'").unwrap(), ("", "\\u{A}"));
+        assert_eq!(char_lit("'\\u{10}'").unwrap(), ("", "\\u{10}"));
+        assert_eq!(char_lit("'\\u{aa}'").unwrap(), ("", "\\u{aa}"));
+        assert_eq!(char_lit("'\\u{10FFFF}'").unwrap(), ("", "\\u{10FFFF}"));
+
+        // Should fail.
+        assert!(char_lit("''").is_err());
+        assert!(char_lit("'\\o'").is_err());
+        assert!(char_lit("'\\x'").is_err());
+        assert!(char_lit("'\\x1'").is_err());
+        assert!(char_lit("'\\x80'").is_err());
+        assert!(char_lit("'\\u'").is_err());
+        assert!(char_lit("'\\u{}'").is_err());
+        assert!(char_lit("'\\u{110000}'").is_err());
+    }
```

Comment:
Can you try with some multi-byte characters, like emoji or Chinese/Japanese code points? `anychar` is called that, but the documentation suggest it actually consumes bytes.

Reply:
Luckily for us you thought about that: I was using `.len()` instead of `.chars().count()`, leading to a wrong result. Fixed it.